dvol_python/test_dvol.py

```diff
@@ -55,43 +80,52 @@ class VoluminousTests(TestCase):
     def setUp(self):
         self.tmpdir = FilePath(self.mktemp())
         self.tmpdir.makedirs()
-        self.patch(Voluminous, "lockFactory", NullLock)
 
     def test_create_volume(self):
-        # TODO test volume names with '/' in them - they should not end up
-        # making nested heirarchy
         dvol = VoluminousOptions()
-        dvol.parseOptions(["-p", self.tmpdir.path, "init", "foo"])
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo"])
         self.assertTrue(self.tmpdir.child("foo").exists())
         self.assertTrue(self.tmpdir.child("foo").child("branches")
                 .child("master").exists())
-        self.assertEqual(dvol.voluminous.getOutput(),
-                ["Created volume foo", "Created branch foo/master"])
+        self.assertEqual(dvol.voluminous.getOutput()[-1],
+                "Created volume foo\nCreated branch foo/master")
 
     def test_create_volume_already_exists(self):
         dvol = VoluminousOptions()
-        dvol.parseOptions(["-p", self.tmpdir.path, "init", "foo"])
-        self.assertRaises(VolumeAlreadyExists,
-                dvol.parseOptions, ["-p", self.tmpdir.path, "init", "foo"])
-        self.assertEqual(dvol.voluminous.getOutput(),
-                ["Error: volume foo already exists"])
+        # Create the repository twice, second time should have the error
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo"])
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo"])
+        self.assertEqual(dvol.voluminous.getOutput()[-1],
+                "Error: volume foo already exists")
+
+    def test_create_volume_with_path_seperator(self):
+        dvol = VoluminousOptions()
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo/bar"])
+        self.assertEqual(dvol.voluminous.getOutput()[-1],
+                "Error: foo/bar is not a valid name")
```

Comment:
I agree this is a good idea but it seems there is a whole raft of improvements that could be made to the test_create_volume* tests to use hypothesis in general, and improve the definition of valid and invalid names. Can we do that in a separate issue/branch to avoid this one becoming too broad in scope?